src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/RestSqlAction.java

```diff
@@ -114,7 +124,7 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
             } else {
                 Metrics.getInstance().getNumericalMetric(MetricName.FAILED_REQ_COUNT_SYS).increment();
             }
-            LOG.error(String.format("[%s] Failed during query execution", sqlRequest.getId()), e);
+            LOG.error(String.format(Locale.ROOT, "[%s] Failed during query execution", LogUtils.getRequestId()), e);
```

Comment:
Just want to learn, what's the difference between this `ROOT` locale and default locale used by `String.format`? Any concern here?

Reply:
ROOT is the location/language/culture-neutral locale, which means if we are using it, the code will behave the same were independent of environment. If you do not specify the `Locale`, it picks the default one, and default means JVM just gets it from the underlying OS settings. That in its turn means if I set my windows location to Armenia and language to Armenian, and you set to US and English, we might get different result from the `String.format` call. Of course we usually don't use special characters which depend on locale, but it's a good practice to use `Locale.ROOT` always (not for strings exposed on the UI) to be on the safe side.

Reply:
> ROOT is the location/language/culture-neutral locale, which means if we are using it, the code will behave the same were independent of environment. If you do not specify the `Locale`, it picks the default one, and default means JVM just gets it from the underlying OS settings. That in its turn means if I set my windows location to Armenia and language to Armenian, and you set to US and English, we might get different result from the `String.format` call. Of course we usually don't use special characters which depend on locale, but it's a good practice to use `Locale.ROOT` always (not for strings exposed on the UI) to be on the safe side.

Good to know this. Thanks!